test/v3/Alchemist.test.js

```diff
@@ -0,0 +1,937 @@
+                await alchemist.connect(minter).mint(parseEther('1000'));
+                await transmuterContract.connect(minter).stake(parseEther('1000'));
+                await alchemist.connect(minter).liquidate(liqAmt);
+                const alchemistTokenBalPost = await token.balanceOf(alchemist.address);
+
+                const transmuterEndingTokenBal = await token.balanceOf(
+                    transmuterContract.address
+                );
+                expect(alchemistTokenBalPost).equal(dep2Amt.sub(liqAmt));
+                expect(transmuterEndingTokenBal).equal(liqAmt);
+            });
+            it('deposits, mints alUsd, repays, and has no outstanding debt', async () => {
+                await alchemist.connect(minter).deposit(depositAmt.sub(parseEther('1000')));
+                await alchemist.connect(minter).mint(mintAmt);
+                await transmuterContract.connect(minter).stake(mintAmt);
+                await alchemist.connect(minter).repay(mintAmt, 0);
+                expect(
+                    await alchemist.connect(minter).getCdpTotalDebt(await minter.getAddress())
+                ).equal(0);
+            });
+            it('deposits, mints, repays, and has no outstanding debt', async () => {
+                await alchemist.connect(minter).deposit(depositAmt);
+                await alchemist.connect(minter).mint(mintAmt);
+                await alchemist.connect(minter).repay(0, mintAmt);
+                expect(
+                    await alchemist.connect(minter).getCdpTotalDebt(await minter.getAddress())
+                ).equal(0);
+            });
+            it('deposits, mints alUsd, repays with alUsd and DAI, and has no outstanding debt', async () => {
+                await alchemist.connect(minter).deposit(depositAmt.sub(parseEther('1000')));
+                await alchemist.connect(minter).mint(mintAmt);
+                await transmuterContract.connect(minter).stake(parseEther('500'));
+                await alchemist.connect(minter).repay(parseEther('500'), parseEther('500'));
+                expect(
+                    await alchemist.connect(minter).getCdpTotalDebt(await minter.getAddress())
+                ).equal(0);
+            });
+
+            it('deposits and liquidates DAI', async () => {
+                await alchemist.connect(minter).deposit(depositAmt);
+                await alchemist.connect(minter).mint(mintAmt);
+                await transmuterContract.connect(minter).stake(mintAmt);
+                await alchemist.connect(minter).liquidate(mintAmt);
+                expect(
+                    await alchemist
+                        .connect(minter)
+                        .getCdpTotalDeposited(await minter.getAddress())
+                ).equal(depositAmt.sub(mintAmt));
+            });
+        });
+
+        // describe('mint', () => {
+        //     let depositAmt = parseEther('5000');
```

Comment:
The test for mint() is commented out.
While it is extra important because it includes the borrowFee now, so some additional test will be necessary.

Reply:
There is issue for setting whitelist in Transmuter. I cannot figure out why it is failing in the test

Reply:
I do see you have an additional setWhitelist that isn't in the original.


https://github.com/yaxis-project/metavault/blob/2ae878edafa305ee053a6105085029639e8fdcc1/test/v3/Alchemist.test.js#L798-L800


https://github.com/alchemix-finance/alchemix-protocol/blob/eec27cfd4a2540816ef9cbe3297fa1e0871237d1/test/contracts/Alchemist.spec.ts#L811-L812

Reply:
yes. I added it for debugging because `setWhitelist ` is not working at line 811 https://github.com/yaxis-project/metavault/blob/2ae878edafa305ee053a6105085029639e8fdcc1/test/v3/Alchemist.test.js#L811